### PR TITLE
Add s3 PutObject/GetObject functions.

### PR DESCRIPTION
**Pull request by chenln1124**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: demos/http/http_demo_s3_download/demo_config.h [hunk 1]
```diff
@@ -78,6 +78,10 @@
     #define ROOT_CA_CERT_PATH    "certificates/AmazonRootCA1.crt"
 #endif
 
+#ifndef ROOT_CA_CERT_PATH_S3
```
Comment (awsblake) [inline demos/http/http_demo_s3_download/demo_config.h]: why we cannot just use ROOT_CA_CERT_PATH?

File: demos/http/http_demo_s3_download/http_demo_s3_download.c [hunk 2]
```diff
@@ -73,6 +73,11 @@
     #define USER_BUFFER_LENGTH    ( 4096 )
 #endif
 
+/* Pointer to the data to upload.*/
+#ifndef DEMO_HTTP_UPLOAD_DATA
+    #define DEMO_HTTP_UPLOAD_DATA    "Hello World!"
```
Comment (awsblake) [inline demos/http/http_demo_s3_download/http_demo_s3_download.c]: The demo can be put the upload object path in here. open the file path, read the file and upload to S3. This seems to me is create a file and the content is Hello World.

File: demos/http/http_demo_s3_download/http_demo_s3_download.c [hunk 3]
```diff
@@ -1647,10 +1937,27 @@ int main( int argc,
...
+                /* Verify the file exists by retrieving the file size. */
+                ret = verifyS3ObjectFileSize( &transportInterface,
+                                              pPath );
+                returnStatus = ( ret == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
+            }
+
+			/******************** Download S3 Object File. **********************/
             if( returnStatus == EXIT_SUCCESS )
             {
                 ret = downloadS3ObjectFile( &transportInterface,
```
Comment (awsblake) [inline demos/http/http_demo_s3_download/http_demo_s3_download.c:2034]: download the file to the local. we can do md5sum check and see the file is same to the local one.

File: demos/http/http_demo_s3_download/demo_config.h (hunk 1 above)
Comment (chenln1124) [reply demos/http/http_demo_s3_download/demo_config.h]: ROOT_ CA_ CERT_ PATH_ S3 is only for compilation and is actually useless. It can be removed.

**Pull request merged**